src/tools/genpolicy/rules.rego

```diff
@@ -754,10 +754,18 @@ allow_var(p_process, i_process, i_var, s_name) {
 # Match input with one of the policy variables, after substituting $(sandbox-name).
 allow_var(p_process, i_process, i_var, s_name) {
     some p_var in p_process.Env
-    p_var2 := replace(p_var, "$(sandbox-name)", s_name)
+    print("allow_var 2: p_var =", p_var)
 
-    print("allow_var 2: p_var2 =", p_var2)
-    p_var2 == i_var
+    p_var_split := split(p_var, "=")
+    count(p_var_split) == 2
+
+    p_var_split[1] == "$(sandbox-name)"
```

Comment:
This seems more restrictive than what we had before - is that intentional to cover only the downward API use case

```yaml
valueFrom:
  fieldRef:
    fieldPath: metadata.name
```